qemu/tests/win_virtio_driver_installer_uninstall.py

```diff
@@ -83,21 +85,17 @@ def run(test, params, env):
     error_context.context("Check if all drivers are uninstalled.", test.log.info)
     # Wait a moment to check if drivers were uninstalled totally
     time.sleep(5)
+
     uninstalled_device = []
-    device_name_list = [
-        "VirtIO RNG Device",
-        "VirtIO Serial Driver",
-        "VirtIO Balloon Driver",
-        "QEMU PVPanic Device",
-        "VirtIO Input Driver",
-        "Red Hat VirtIO Ethernet Adapter",
-        "VirtIO FS Device",
-        "QEMU FwCfg Device",
+
+    # Filter out VirtIO Viomem Driver if boot_with_viomem is not enabled
+    if params.get("boot_with_viomem", "no") == "no":
+        no_need_uninstall_dev_list.append("VirtIO Viomem Driver")
```

Comment:
I noticed an AttributeError waiting to happen on this line. Python strings are immutable, meaning they can't be changed in place, so they don't support the .append() method like lists do.
I would suggest to convert string to list first, and then call the append method to add items to the list
`ast.literal_eval()`

Reply:
Updated, thanks. @leidwang